File: ios-client/StripeApplication/StripeApplication/Data/Models/PaymentMethod.swift
import Foundation

struct CardDetails: Codable {
    let brand: String
    let last4: String
    let expMonth: Int
    let expYear: Int

    enum CodingKeys: String, CodingKey {
        case brand, last4
        case expMonth = "exp_month"
        case expYear = "exp_year"
    }
}

struct PaymentMethod: Codable, Identifiable {
    let id: String
    let type: String
    let card: CardDetails?
    let created: Int64
}

struct PaymentMethodsResponse: Codable {
    let paymentMethods: [PaymentMethod]

    enum CodingKeys: String, CodingKey {
        case paymentMethods = "payment_methods"
    }
}

struct SetupIntentResponse: Codable {
    let clientSecret: String
    let setupIntentId: String
    let customerId: String?

    enum CodingKeys: String, CodingKey {
        case clientSecret = "client_secret"
        case setupIntentId = "setup_intent_id"
        case customerId = "customer_id"
    }
}

File: android-client/app/src/main/java/com/example/stripedemo/data/models/PaymentMethod.kt
package com.example.stripedemo.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CardDetails(
    val brand: String,
    val last4: String,
    @Json(name = "exp_month")
    val expMonth: Int,
    @Json(name = "exp_year")
    val expYear: Int
)

@JsonClass(generateAdapter = true)
data class PaymentMethod(
    val id: String,
    val type: String,
    val card: CardDetails?,
    val created: Long
)

@JsonClass(generateAdapter = true)
data class PaymentMethodsResponse(
    @Json(name = "payment_methods")
    val paymentMethods: List<PaymentMethod>
)

@JsonClass(generateAdapter = true)
data class SetupIntentResponse(
    @Json(name = "client_secret")
    val clientSecret: String,
    @Json(name = "setup_intent_id")
    val setupIntentId: String,
    @Json(name = "customer_id")
    val customerId: String? = null
)
